src/bindings/python/tests/test_runtime/test_core.py

```diff
@@ -77,24 +69,57 @@ def test_core_class(device):
 
 
 # request - https://docs.pytest.org/en/7.1.x/reference/reference.html#request
-def test_compile_model(request, tmp_path, device):
+@pytest.mark.parametrize("device_name", [
+    None,
+    "CPU"
+])
+def test_compile_model(request, tmp_path, device_name):
     core = Core()
     xml_path, bin_path = create_filename_for_test(request.node.name, tmp_path)
     relu_model = get_relu_model()
     serialize(relu_model, xml_path, bin_path)
     model = core.read_model(model=xml_path, weights=bin_path)
-    compiled_model = core.compile_model(model, device)
+    compiled_model = None
+    if device_name is None:
+        compiled_model = core.compile_model(model)
+    else:
+        compiled_model = core.compile_model(model, device_name)
+
     assert isinstance(compiled_model, CompiledModel)
 
 
-# request - https://docs.pytest.org/en/7.1.x/reference/reference.html#request
-def test_compile_model_without_device(request, tmp_path):
-    core = Core()
-    xml_path, bin_path = create_filename_for_test(request.node.name, tmp_path)
+@pytest.mark.parametrize("loading_type", [
+    None,
+    "posixpath",
+    "auto"
+])
+@pytest.mark.parametrize("device_name", [
+    None,
+    "CPU"
+])
+@pytest.mark.parametrize("config", [
+    None,
+    {hints.performance_mode(): hints.PerformanceMode.THROUGHPUT},
+    {hints.performance_mode(): hints.PerformanceMode.LATENCY},
+    {hints.performance_mode(): hints.PerformanceMode.CUMULATIVE_THROUGHPUT}
+])
+def test_compact_api(request, tmp_path, loading_type, device_name, config):
```

Comment:
```suggestion
@pytest.fixture
def get_model():
    return get_relu_model()


@pytest.fixture
def get_model_path(request, tmp_path):
    xml_path, _ = create_filename_for_test(request.node.name, tmp_path, True)
    serialize(get_relu_model(), xml_path)
    return Path(xml_path)


@pytest.mark.parametrize("model_type", [
    "get_model",
    "get_model_path",
])
@pytest.mark.parametrize("device_name", [
    None,
    "CPU"
])
@pytest.mark.parametrize("config", [
    None,
    {"PERFORMANCE_HINT": "THROUGHPUT"},
])
def test_compact_api(model_type, device_name, config, request):
```

Reply:
Should we be using `{hints.performance_mode(): hints.PerformanceMode.THROUGHPUT}` or `{"PERFORMANCE_HINT": "THROUGHPUT"}`?